treeherder/webapp/api/jobs.py

```diff
@@ -96,9 +97,80 @@ class Meta:
         }
 
 
-class JobsViewSet(viewsets.ViewSet):
+class JobPagination(PageNumberPagination):
```

Comment:
I don't know if this is still an issue, but the Django Paginator was using the `count` method, performing a full table scan so you might find this useful: https://github.com/mozilla/treeherder/pull/4039

I ended up removing the custom paginator at a later point because of feature requests that made it easier to serve all results at once, so that custom pagination isn't in the code base anymore.